src/main/java/de/unistuttgart/iste/gits/content_service/controller/SubscriptionController.java

```diff
@@ -51,7 +53,36 @@ public Mono<Void> logUserProgress(@RequestBody CloudEvent<UserProgressLogEvent>
     @Topic(name = "chapter-changes", pubsubName = "gits")
     @PostMapping(path = "/content-service/chapter-changes-pubsub")
     public Mono<Void> cascadeCourseDeletion(@RequestBody CloudEvent<ChapterChangeEvent> cloudEvent, @RequestHeader Map<String, String> headers) {
+        if (cloudEvent == null || cloudEvent.getData() == null) {
+            return Mono.error(new IllegalArgumentException("CloudEvent or ChapterChangeEvent is null"));
```

Comment:
Please no Mono.error (will cause infinite loops with it happens for some reason)
just log it